### ISSUE TITLE
Images - Changing images without renaming

### ISSUE THREAD

**Issue by bradystroud** [tinacms/tinacms#3820]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

cc @BrookJeynes @harry-Ross
When I update an image, but the name doesn't change, TinaCMS doesn't notice anything has changed and uses the old image.

### Reproduction

https://github.com/SSWConsulting/SSW.Website-v3

### Steps to reproduce

1. Add an image in the CMS
2. Replace the image but keep the name the same
3. Change will appear to have worked locally, but once deployed the changes are not visible

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (8) x64 Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz
    Memory: 65.65 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.14.0 - ~/.nvm/versions/node/v18.14.0/bin/node
    Yarn: 1.22.19 - ~/Developer/SSW/SSW.Website-v3/node_modules/.bin/yarn
    npm: 9.3.1 - ~/.nvm/versions/node/v18.14.0/bin/npm
  Browsers:
    Brave Browser: 109.1.47.186
    Chrome: 112.0.5615.49
    Firefox: 109.0.1
    Safari: 16.4
  npmPackages:
    @tinacms/cli: ^1.5.3 => 1.5.3 
    tinacms: ^1.4.6 => 1.4.3
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by logan-anderson** [comment]
@bradystroud This is currently a limitation of our git backed media solution.  In order to not rely on when your sites deployed, in production we serve a copy of your sites images from our own asset server. Our asset server is not aware of branches so we cannot override existing media (If someone pushes an update to a feature branch that updates an image we don't want to update the production copy). If you want to change the image you should be able to delete the image from the media manager UI in production and then click "Sync Media". The new image should appear. 

We are thinking and working on a better solution (and are open to suggestions) but this is the current behaviour

**Comment by bradystroud** [comment]
This is still an issue and should not be closed.

At a minimum, could this known limitation be documented somewhere?

**Comment by bradystroud** [comment]
I really think this is important to document somewhere

**Comment by zackjenks** [comment]
We're having the same issue. Even after deleting the image from the library and resyncing, the new image is not loading. Has there been any progress on this?

**Comment by andrewwaltosssw** [comment]
Experienced this issue whist trying to fix another issue #4465.
